test/tools/unified-spec-runner/entities.ts

```diff
@@ -48,6 +51,35 @@ export interface UnifiedChangeStream extends ChangeStream {
   eventCollector: InstanceType<typeof import('../../tools/utils')['EventCollector']>;
 }
 
+export class UnifiedThread {
+  #promise: Promise<void>;
+  #error: Error;
+  #killed = false;
+
+  id: string;
+
+  constructor(id) {
+    this.id = id;
+    this.#promise = Promise.resolve();
+  }
+
+  queue(functionToQueue: () => Promise<any>) {
+    if (this.#killed || this.#error) {
+      return;
+    }
+
+    this.#promise = this.#promise.then(functionToQueue).catch(e => (this.#error = e));
```

Comment:
because we don't want an unhandled rejection before we are ready to await the result of the promise chain; each newly queued function gets executed immediately, and this entire line is making an unawaited but very much executed promise (per the spec, we want to execute the operations immediately on the "thread", not waiting for all operations to be stacked)

good suggestion on the docs though